llvm/tools/llvm-objdump/MachODump.cpp

```diff
@@ -3673,6 +3683,12 @@ struct method_list32_t {
   /* struct method32_t first;  These structures follow inline */
 };
 
+struct method_list_delta_t {
+  uint32_t entsize;
+  uint32_t count;
+  /* struct method_delta_t first;  These structures follow inline */
+};
+
```

Comment:
In my opinion, I think `method_list_delta_t` is not necessary, and might be confusing. Since `print_method_list_delta_t` is the first to run, one might have the message about `method_delta_t extends past the end of the section` before it has been confirmed it was a `method_delta_t`.

I think it might be better to let the first block of `print_method_list32_t` happen (old lines 4538 to 4554) until one can check against `ML_HAS_DELTAS`, and then diverge from there (passing `ml` to the child methods).

Same change in `print_method_list64_t`. One can go down to line 4463 before diverging to the special case for delta handling.

Reply:
This is an option, but when I looked at it I remember it would result in overly-inter-linked code. 
What if we just bail out `print_method_list_delta_t` before confirming it's an actual delta list ? 
Then there would be no confusing message printed out about `method_delta_t extends past the end of the section`. ?

Reply:
If you notice in `print_method_list32_t` and `print_method_list64_t` they already compromise and print "method_list_t entends [sic] past the end of the section", not `method_list32_t` or `method_list64_t`. Those two are actually the same struct, with different names (the only difference is if they keep `method32_t`, `method64_t`, or the new `method_delta_t`).

I think the code doesn't interlink too much, but that's an opinion:

```
static void print_method_list64_t(uint64_t p, struct DisassembleInfo *info,
                                const char *indent) {
  struct method_list64_t ml;
  struct method64_t m;
  const char *r;
  uint32_t offset, xoffset, left, i;
  SectionRef S, xS;
  const char *name, *sym_name;
  uint64_t n_value;

  r = get_pointer_64(p, offset, left, S, info);
  if (r == nullptr)
    return;
  memset(&ml, '\0', sizeof(struct method_list64_t));
  if (left < sizeof(struct method_list64_t)) {
    memcpy(&ml, r, left);
    outs() << "   (method_list_t entends past the end of the section)\n";
  } else
    memcpy(&ml, r, sizeof(struct method_list64_t));
  if (info->O->isLittleEndian() != sys::IsLittleEndianHost)
    swapStruct(ml);

  if ((ml.entsize & ML_HAS_DELTAS) != 0) {
    print_method_list_delta_t(ml, p, info, indent);
    return;
  }
  
  // The rest of the code
```

Reply:
Yes, but now we're alternatively passing `method_list64_t` / `method_list32_t` to `print_method_list_delta_t` - so we have to specialize it, or hard case to `method_list_delta_t` which is OK since they're same structure layout, but can be confusing. I think the single line at the top offers best separation / layout (and making sure nothing is printed until we confirm delta list). 
Or, do you see a better solution for passing the `method_list64_t` / `method_list32_t` to `print_method_list_delta_t` ?

Reply:
Or you can pass the entsize and count and individual values. Passing a struct of two `uint32_t` is going to be almost the same code, after all: `print_method_list_delta_t(ml.entsize, ml.count, p, info, indent);`

Reply:
Updated to passing `ml.entsize, ml.count`